tests/test_strategies.py

```diff
@@ -0,0 +1,232 @@
+            "vehicle_events": [],
+        }
+    }
+
+
+def load_json(filename):
+
+    with open(filename, 'r') as f:
+        return json.load(f)
+
+
+class TestScenarios(unittest.TestCase):
+
+    def test_scenario_times(self):
+        # correct number of timesteps?
+        j = get_test_json()
+        s = scenario.Scenario(j)
+        self.assertEqual(s.n_intervals, 96)
+
+        # either n_intervals or stop time, not both
+        j['scenario']['stop_time'] = "2020-01-01T01:00:00+02:00"
+        with self.assertRaises(AssertionError):
+            s = scenario.Scenario(j)
+
+        # remove n_intervals, stop time remains: success, four timesteps
+        del j['scenario']['n_intervals']
+        s = scenario.Scenario(j)
+        self.assertEqual(s.n_intervals, 4)
+
+    def test_file(self):
+        # open from file
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        scenario.Scenario(load_json(input), os.path.dirname(input))
+
+    # TEST SCENARIOS WITH BATTERY, FEEDIN AND EXTERNAL LOAD (Scenario A)
+
+    def test_greedy_A(self):
+        # test basic strategy
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('greedy', {})
+
+    def test_balanced_A(self):
+        # test basic strategy
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('balanced', {})
+
+    def test_balanced_market_A(self):
+        # test basic strategy
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('balanced_market', {})
+
+    def test_flex_window_A(self):
+        # test basic strategy
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('flex_window', {})
+
+    def test_peak_load_window_A(self):
+        # test basic strategy
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_A.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('peak_load_window', {})
+
+    # TEST with battery, feedin, extLoad and V2G (Scenario B)
+
+    def test_greedy_B(self):
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_B.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('greedy', {})
+
+    def test_balanced_B(self):
+        input = os.path.join(TEST_REPO_PATH, 'test_data/input_test_strategies/scenario_B.json')
+        s = scenario.Scenario(load_json(input), os.path.dirname(input))
+        s.run('balanced', {})
```

Comment:
Is there something that can be tested along the way, e.g., if balanced allocates the same energy for every timestep?

Reply:
yes, probably. I only added some specific tests for flex_window, but more should and will follow for the other strategies.